lib/encoding/url.toit

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+NEEDS_ENCODING_ ::= ByteArray '~' - '-' + 1:
+  c := it + '-'
+  (c == '-' or c == '_' or c == '.' or c == '~' or '0' <= c <= '9' or 'A' <= c <= 'Z' or 'a' <= c <= 'z') ? 0 : 1
+
+// Takes an ASCII string or a byte array.
```

Comment:
Maybe add that it returns the number of escapes needed? Or change the name to `count_escapes_`?